host/xtest/regression_1100.c

```diff
@@ -0,0 +1,496 @@
+		{ .type = TEEC_SESSION_SETTING_DATA,
+		  .u.data = &data_setting }
+	};
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_InitializeContext2(xtest_tee_name, &ctx, ctx_settings, 1)))
+		goto no_ctx;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_OpenSession2(&ctx, &sess, &ocall_ta_uuid,
+				  TEEC_LOGIN_PUBLIC, NULL, NULL, &ret_orig,
+				  sess_settings, 1)))
+		goto no_sess;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c,
+		TEE_ERROR_GENERIC,
+		TEEC_InvokeCommand(&sess,
+				   TA_OCALL_CMD_PREMATURE_CONTEXT_FINALIZE,
+				   NULL, &ret_orig)))
+		goto no_invoke;
+
+	ret = true;
+
+	/* The context will have been finalized by the OCALL handler */
+	goto no_ctx;
+
+no_invoke:
+	TEEC_CloseSession(&sess);
+no_sess:
+	TEEC_FinalizeContext(&ctx);
+no_ctx:
+	return ret;
+}
+
+static bool test_ocall_during_session_open(ADBG_Case_t *c)
+{
+	TEEC_Context ctx = {};
+	TEEC_Session sess = {};
+	uint32_t ret_orig = 0;
+	TEEC_Operation op = { 0 };
+	bool ret = false;
+
+	TEEC_ContextSettingOcall ocall_setting = { ocall_handler, &ctx };
+	TEEC_ContextSetting ctx_settings[] = {
+		{ .type = TEEC_CONTEXT_SETTING_OCALL,
+		  .u.ocall = &ocall_setting }
+	};
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_InitializeContext2(xtest_tee_name, &ctx, ctx_settings, 1)))
+		return false;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_INPUT, TEEC_NONE,
+					 TEEC_NONE, TEEC_NONE);
+	op.params[0].value.a = TA_OCALL_CMD_OPEN_SESSION_OCALL;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_OpenSession(&ctx, &sess, &ocall_ta_uuid, TEEC_LOGIN_PUBLIC,
+				 NULL, &op, &ret_orig)))
+		goto no_sess;
+
+	/* Executes after the OCALL handler */
+	ADBG_EXPECT_BOOLEAN(c, true, session_open_ocall_reply);
+
+	TEEC_CloseSession(&sess);
+	ret = true;
+
+no_sess:
+	TEEC_FinalizeContext(&ctx);
+	return ret;
+}
+
+static bool text_ctx_finalize_during_session_open_ocall(ADBG_Case_t *c)
+{
+	TEEC_Context ctx = {};
+	TEEC_Session sess = {};
+	uint32_t ret_orig = 0;
+	TEEC_Operation op = { 0 };
+	bool ret = false;
+
+	TEEC_ContextSettingOcall ocall_setting = { ocall_handler, &ctx };
+	TEEC_ContextSetting ctx_settings[] = {
+		{ .type = TEEC_CONTEXT_SETTING_OCALL,
+		  .u.ocall = &ocall_setting }
+	};
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_InitializeContext2(xtest_tee_name, &ctx, ctx_settings, 1)))
+		return false;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_INPUT, TEEC_NONE,
+					 TEEC_NONE, TEEC_NONE);
+	op.params[0].value.a =
+		TA_OCALL_CMD_OPEN_SESSION_OCALL_PREMATURE_CONTEXT_FINALIZE;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_ERROR_GENERIC,
+		TEEC_OpenSession(&ctx, &sess, &ocall_ta_uuid, TEEC_LOGIN_PUBLIC,
+				 NULL, &op, &ret_orig)))
+		goto no_sess;
+
+	/* Executes after the OCALL handler */
+	ADBG_EXPECT_BOOLEAN(c, true,
+			    session_open_ocall_premature_ctx_finalize_reply);
+
+	/*
+	 * The OCALL handler will have closed the session by finalizing the
+	 * context.
+	 */
+	ret = true;
+
+no_sess:
+	TEEC_FinalizeContext(&ctx);
+	return ret;
+}
+
+static void xtest_tee_test_1100(ADBG_Case_t *c)
```

Comment:
For consistency with all other series this should be `_1101` (but I'm not sure there is a rational reason TBH)

Reply:
Fixed.